vault/templates/secret.yaml

```diff
@@ -1,24 +1,27 @@
 {{- if and (not .Values.tls.secretName) (not .Values.certManager.certificate.enabled) }}
-{{- $cn := include "vault.fullname" . -}}
+{{- $cn := include "vault.fullname" $ -}}
 {{- $externalDNS := print $cn "." .Release.Namespace -}}
 {{- $ca := genCA "vault-ca" 365 -}}
 {{- $cert := genSignedCert $cn (list "127.0.0.1") (list $cn $externalDNS) 365 $ca -}}
+{{- range $i, $namespace := .Values.tls.namespaces }}
```

Comment:
There should be an if-else block to keep the original behaviour, i.e. if no namespace is specified then use the release namespace, to avoid a breaking change!

Reply:
Or even better, the secret should be created in the release namespace every time, and besides this, a list of extra namespaces could be specified, similar to what is implemented in the Bank-Vaults operator: https://github.com/bank-vaults/vault-operator/blob/e931415a8ac9b474658e7d0c9a5f0366b9f9026f/deploy/examples/cr.yaml#L92

Reply:
Hi, I can work on the first suggestion. Could you help me with how the second one (the ideal one) is implemented? That is what I was trying to replicate in the first place.

Reply:
In the `vault-operator` this logic is implemented in the operator code, here is one way to achieve a similar result:

```
...
{{- $caNamespaces := list .Release.Namespace -}}
{{- if .Values.tls.caNamespaces }}{{ $caNamespaces = (concat $caNamespaces .Values.tls.caNamespaces ) | uniq }}{{ end -}}
{{- range $i, $namespace := $caNamespaces }}
...
```

Here the first step is to create a one element array from the release namespace since we would like to include it by all means, then if there is anything defined in `tls.caNamespaces` (I changed it from `namespaces` to match the config in the operator) join those elements as well, call `uniq` to weed out duplicates and lastly iterate on the list and create the secrets.

Reply:
Thanks! Tested it locally and it works, updated with the changes. :+1:

Just curious, is there any way we could make it work also for `["*"]` (i.e. all namespaces), like in the Vault operator?

Reply:
I think helm is unaware of existing namespaces in a cluster so this is as far as we can get...